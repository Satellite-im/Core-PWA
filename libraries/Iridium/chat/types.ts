export type ConversationMessage = {
  conversationId: string
  from: string
  body: string
  at: number
  type: 'text' | 'file' | 'group' | 'glyph' | 'divider' | 'string'
  attachments: string[]
}

export type Conversation = {
  id: string
  type: 'direct' | 'group'
  name?: string
  participants: string[]
  createdAt: number
  updatedAt: number
  messages: {
    [key: string]: ConversationMessage
  }
  messageArray: ConversationMessage[]
}

export const ChatError = {
  CONVERSATION_EXISTS: 'error.chat.conversation_exists',
  CONVERSATION_NOT_FOUND: 'error.chat.conversation_not_found',
  MESSAGE_NOT_SENT: 'error.chat.message_not_sent',
}
