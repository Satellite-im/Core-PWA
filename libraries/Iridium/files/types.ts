/* eslint-disable no-use-before-define */
import { DIRECTORY_TYPE } from '~/libraries/Files/types/directory'
import { FILE_TYPE } from '~/libraries/Files/types/file'

export type FileUpdate = 'upload' | 'delete' | 'like' | 'share'

interface Shared {
  id: string
  name: string
  liked: boolean
  shared: boolean
  modified: number
  size: number
  parentId: IridiumDirectory['id'] // empty string if root item
}

export interface IridiumFile extends Shared {
  type: FILE_TYPE
  description?: string
  thumbnail: string
  extension: string
  nsfw: boolean
}

export interface IridiumDirectory extends Shared {
  type: DIRECTORY_TYPE
  children: IridiumItem[]
}

export type IridiumItem = IridiumFile | IridiumDirectory
