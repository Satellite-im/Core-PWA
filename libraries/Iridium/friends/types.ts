import { User } from '../users/types'

export type Friend = User & {}

export type FriendRequestStatus =
  | 'pending'
  | 'accepted'
  | 'rejected'
  | 'removed'

export type FriendRequest = {
  did: User['did']
  incoming: boolean
  status: FriendRequestStatus
  at: number
}

export const FriendsError = {
  NETWORK_ERROR: 'errors.friends.network',
  FRIEND_EXISTS: 'errors.friends.exists',
  FRIEND_NOT_FOUND: 'errors.friends.not_found',
  REQUEST_NOT_FOUND: 'errors.friends.request_not_found',
  REQUEST_NOT_SENT: 'errors.friends.request_not_sent',
  REQUEST_ALREADY_SENT: 'errors.friends.request_already_sent',
}
