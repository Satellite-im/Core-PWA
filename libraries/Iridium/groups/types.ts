import { IridiumDocument } from '@satellite-im/iridium'
import Group from './Group'

export type GroupMemberDetails = {
  id: string
  name: string
  photoHash: string
}

export type GroupConfig = {
  name: string
  origin: string
  avatar?: string
  members: { [did: string]: GroupMemberDetails }
  metadata?: IridiumDocument
}

export type GroupData = GroupConfig & {
  id: string
}

export type GroupMap = { [key: string]: Group }

export type GroupManagerEvent =
  | 'group-created'
  | 'group-updated'
  | 'group-deleted'
  | 'group-member-added'
  | 'group-member-removed'
  | 'group-member-updated'
  | 'group-member-joined'
  | 'group-member-left'

export enum GroupsError {
  NETWORK_ERROR = 'errors.groups.network',
  NOT_A_MEMBER = 'errors.groups.not_a_member',
  GROUP_ALREADY_EXISTS = 'error.groups.group_already_exists',
  GROUP_NOT_FOUND = 'error.groups.group_not_found',
  GROUP_NOT_INITIALIZED = 'errors.groups.group_not_initialized',
  INVITE_ALREADY_SENT = 'errors.groups.request_already_sent',
  RECIPIENT_NOT_FOUND = 'errors.groups.recipient_not_found',
  USER_NOT_INITIALIZED = 'errors.groups.user_not_initialized',
}
