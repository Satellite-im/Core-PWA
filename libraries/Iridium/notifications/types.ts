import { IridiumPeerIdentifier } from '~/../iridium/dist'

export const NotificationsError = {
  NOTIFICATION_NOT_SENT: 'error.notifications.notifications_not_sent',
}

export enum NotificationType {
  FRIEND_REQUEST = 'friend_request',
  DIRECT_MESSAGE = 'direct_message',
  GROUP_MESSAGE = 'group_message',
  MENTION = 'mention',
  MEMBER_JOIN = 'member_join',
  MEMBER_LEAVE = 'member_leave',
  ADDED_TO_GROUP = 'added_to_group',
  GROUP_CONVERSATION_CREATED = 'group_conversation_created',
  CALL_INCOMING = 'call_incoming',
}

export type NotificationTypeValues = NotificationType[keyof NotificationType]

export type Notification<P = {}> = {
  at: number
  type: NotificationType
  senderId: IridiumPeerIdentifier
  title: string
  description: string
  seen: boolean
  payload: P
  id?: string
  image?: string
}

export type FriendRequestNotificationPayload = {
  senderId: IridiumPeerIdentifier
}

export type MessageNotificationPayload = {
  conversationId: string
  messageId: string
}

export type GroupConversationCreatedNotificationPayload = {
  conversationId: string
}

export type MemberJoinNotificationPayload = MessageNotificationPayload & {
  addedMemberIds: IridiumPeerIdentifier[]
}

export enum ActionTypeIds {
  CALL_INCOMING = 'call_incoming',
  DEFAULT = 'default',
  FRIEND_REQUEST = 'friend_request',
  MESSAGE = 'message',
}

export type NotificationPayloads =
  | FriendRequestNotificationPayload
  | GroupConversationCreatedNotificationPayload
  | MemberJoinNotificationPayload
  | MessageNotificationPayload

export type NotificationBase<P = {}> = Omit<
  Notification<P>,
  'title' | 'description'
>

export type NotifCreateHandler<P = {}> = {
  handler: (notif: P) => void
}

export enum NotifActionTypes {
  TAP = 'tap',
  RESPOND = 'respond',
  ACCEPT = 'accept',
  DECLINE = 'decline',
}

export type NotifActionHandler<P = {}> = {
  [key in NotifActionTypes]?: (payload: P, inputValue?: string) => void
}
