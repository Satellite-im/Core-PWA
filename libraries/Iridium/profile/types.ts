import { User } from '../friends/types'

export type Profile = User & {
  friends: Friend[]
  chats: Chat[]
  groups: Group[]
  notifications: Notification[]
  settings: IridiumDocument
}
