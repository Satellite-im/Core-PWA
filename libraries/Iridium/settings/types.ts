import { Sounds } from '~/libraries/SoundManager/SoundManager'
import { hexToRGB } from '~/utilities/Colors'

export enum ThemeKeys {
  DEFAULT = 'default',
  MOONLESS_NIGHT = 'moonlessNight',
}

export const themes: { [key in ThemeKeys]: string } = {
  default: 'Default',
  moonlessNight: 'Moonless Night',
}

export enum FlairKeys {
  SATELLITE = 'satellite',
  PEACH = 'peach',
  PINK = 'pink',
  LIME = 'lime',
  PURPLE = 'purple',
  LAVENDER = 'lavender',
  SUNFLOWER = 'sunflower',
  DEEPBLUE = 'deepBlue',
  VOID = 'void',
}

export type Flair = {
  name: string
  primary: string
  secondary: string
  primaryRGB: string
}

function flairDefinition(flair: Omit<Flair, 'primaryRGB'>): Flair {
  return {
    ...flair,
    primaryRGB: hexToRGB(flair.primary),
  }
}

export const flairs: { [key in FlairKeys]: Flair } = {
  satellite: flairDefinition({
    name: 'Satellite',
    primary: '#2761fd',
    secondary: '#286cfe',
  }),
  peach: flairDefinition({
    name: 'Peach',
    primary: '#ed4c67',
    secondary: '#ed5672',
  }),
  pink: flairDefinition({
    name: 'Pink',
    primary: '#fda7df',
    secondary: '#fdb1e9',
  }),
  lime: flairDefinition({
    name: 'Lime',
    primary: '#a3cb38',
    secondary: '#aed542',
  }),
  purple: flairDefinition({
    name: 'Purple',
    primary: '#6f1e51',
    secondary: '#80215d',
  }),
  lavender: flairDefinition({
    name: 'Lavender',
    primary: '#9980fa',
    secondary: '#a891ff',
  }),
  sunflower: flairDefinition({
    name: 'Sunflower',
    primary: '#ffc312',
    secondary: '#faca3e',
  }),
  deepBlue: flairDefinition({
    name: 'Deep Blue',
    primary: '#30336b',
    secondary: '#2d328a',
  }),
  void: flairDefinition({
    name: 'Void',
    primary: '#2c3a47',
    secondary: '#36434f',
  }),
}

export enum LanguageKeys {
  'EN_US' = 'en_US',
}

export const languages: { [key in LanguageKeys]: string } = {
  en_US: 'English (USA)',
}

export type Settings = {
  theme: ThemeKeys
  flair: FlairKeys
  language: LanguageKeys
  video: {
    flipLocalStream: boolean
  }
  audio: {
    sounds: { [key in Sounds]: boolean }
  }
}
