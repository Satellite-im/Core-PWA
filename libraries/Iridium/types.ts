export type User = {
  id: string
  name: string
  avatar: string
  metadata: { [key: string]: any }
}
