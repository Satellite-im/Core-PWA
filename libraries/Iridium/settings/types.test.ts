import { defaultKeybinds, KeybindKeys, languages, themes } from './types'

describe('themes', () => {
  test('check themes export type', () => {
    expect(themes).toEqual({
      default: 'Default',
      moonlessNight: 'Moonless Night',
    })
  })
})

describe('languages', () => {
  test('check languages export type', () => {
    expect(languages).toEqual({
      en_US: 'English (USA)',
    })
  })
})

describe('defaultKeybinds', () => {
  test('check defaultKeybinds export type', () => {
    expect(defaultKeybinds).toEqual({
      [KeybindKeys.TOGGLE_MUTE]: 'alt+m',
      [KeybindKeys.TOGGLE_DEAFEN]: 'alt+d',
      [KeybindKeys.OPEN_SETTINGS]: 'alt+s',
      [KeybindKeys.CALL_ACTIVE_CHAT]: 'alt+c',
    })
  })
})
