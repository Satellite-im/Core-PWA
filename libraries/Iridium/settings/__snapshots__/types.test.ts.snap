// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Iridium/settings/types should return default keybinds 1`] = `
Object {
  "callActiveChat": "alt+c",
  "openSettings": "alt+s",
  "toggleDeafen": "alt+d",
  "toggleMute": "alt+m",
}
`;

exports[`Test Iridium/settings/types should return default sounds 1`] = `
Object {
  "call": true,
  "connected": true,
  "deafen": true,
  "hangup": true,
  "mute": true,
  "newMessage": true,
  "undeafen": true,
  "unmute": true,
  "upload": true,
}
`;

exports[`Test Iridium/settings/types should return flairs 1`] = `
Object {
  "deepBlue": Object {
    "name": "Deep Blue",
    "primary": "#30336b",
    "primaryRGB": "48, 51, 107",
    "secondary": "#2d328a",
  },
  "lavender": Object {
    "name": "Lavender",
    "primary": "#9980fa",
    "primaryRGB": "153, 128, 250",
    "secondary": "#a891ff",
  },
  "lime": Object {
    "name": "Lime",
    "primary": "#a3cb38",
    "primaryRGB": "163, 203, 56",
    "secondary": "#aed542",
  },
  "peach": Object {
    "name": "Peach",
    "primary": "#ed4c67",
    "primaryRGB": "237, 76, 103",
    "secondary": "#ed5672",
  },
  "pink": Object {
    "name": "Pink",
    "primary": "#fda7df",
    "primaryRGB": "253, 167, 223",
    "secondary": "#fdb1e9",
  },
  "purple": Object {
    "name": "Purple",
    "primary": "#6f1e51",
    "primaryRGB": "111, 30, 81",
    "secondary": "#80215d",
  },
  "satellite": Object {
    "name": "Satellite",
    "primary": "#2761fd",
    "primaryRGB": "39, 97, 253",
    "secondary": "#286cfe",
  },
  "sunflower": Object {
    "name": "Sunflower",
    "primary": "#ffc312",
    "primaryRGB": "255, 195, 18",
    "secondary": "#faca3e",
  },
  "void": Object {
    "name": "Void",
    "primary": "#2c3a47",
    "primaryRGB": "44, 58, 71",
    "secondary": "#36434f",
  },
}
`;

exports[`Test Iridium/settings/types should return languages 1`] = `
Object {
  "en_US": "English (USA)",
}
`;

exports[`Test Iridium/settings/types should return themes 1`] = `
Object {
  "default": "Default",
  "moonlessNight": "Moonless Night",
}
`;
